File: totosuki/ABC/A/024.py
A, B, C, K = map(int, input().split())
S, T = map(int, input().split())
rslt = 0
if S+T >= K:
  rslt += A*S - C*S
  rslt += B*T - C*T
  print(rslt)
else:
  rslt += A*S + B*T
  print(rslt)

File: OCH_Tao/ABC/A/024.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const ABCK = (input[0].split(" ")).map(str => parseInt(str, 10));
const ST = (input[1].split(" ")).map(str => parseInt(str, 10));
let x = ABCK[0] * ST[0] + ABCK[1] * ST[1];
if (ABCK[3] <= ST[0] + ST[1]) {
  x -= ABCK[2] * (ST[0] + ST[1]);
}
console.log(x);